src/components/SnapshotDetailView.jsx

```diff
@@ -141,31 +141,40 @@ export class SnapshotDetailView extends React.PureComponent {
     const datasets = snapshot && snapshot.source && snapshot.source.map((s) => s.dataset);
 
     return (
-      <div id="snapshot-detail-view" className={classes.wrapper}>
-        <div className={classes.width}>
-          <DetailViewHeader
-            of={snapshot}
-            stewards={snapshotStewards}
-            addSteward={this.addSteward}
-            removeSteward={this.removeSteward}
-            readers={snapshotReaders}
-            addReader={this.addReader}
-            removeReader={this.removeReader}
-            terraUrl={terraUrl}
-            canReadPolicies={canReadPolicies}
-            dispatch={dispatch}
-            userRoles={userRoles}
-            user={user}
-          />
-          {snapshot && snapshot.source && (
-            <DatasetTable
-              datasets={filteredDatasets || datasets}
-              datasetsCount={snapshot.source.length}
-              features={user.features}
-              handleFilterDatasets={this.handleFilterDatasets}
-              searchString=""
+      <div className={classes.pageRoot}>
+        <AppBreadcrumbs
```

Comment:
Again, I think this is simplified if you just pass in the breadcrumb elements

Reply:
I don't think it would be simpler. By passing in elements directly, we'd be letting each page be able to potentially style the breadcrumbs itself. Additionally, the page doesn't necessarily know where it is in the hierarchy, so containing the logic in `AppBreadcrumbs` means we don't have to continually calculate the url for hrefs. Also, if we refactor again, we have to go through and make sure that every invocation of `AppBreadcrumbs` is updated to reflect a new hierarchy. By containing all the logic in `AppBreadcumbs`, we avoid duplication and make refactoring easier.